### ISSUE TITLE
Hang on reboot

### ISSUE THREAD

**Issue by terminatorul** [terminatorul/NvStrapsReBar#35]
@TheUntouchable
> GPU: Gigabyte RTX 2070 SUPER GAMING OC 8G
> CPU: AMD Ryzen 7 5800X3D
> Motherboard: MSI X470 GAMING PRO CARBON
> 
> Since I activated the BAR thing I can't reboot anymore. The PC will hang after the operating system is powered off, have to press the power button for some seconds and start the pc again.
> 
> Also the new Nvidia App (Beta) is not detecting the BAR even after reinstalling the drive completely.
> 
> ![grafik](https://github.com/terminatorul/NvStrapsReBar/assets/2162540/c77aafbb-8b26-4b77-9df1-b5008c103d90)
> ![grafik](https://github.com/terminatorul/NvStrapsReBar/assets/2162540/a410e91b-826a-400e-87af-1af4ff38afbd)
> ![grafik](https://github.com/terminatorul/NvStrapsReBar/assets/2162540/d36cc3c9-6543-4850-9515-85c32193d6c7)
> ![grafik](https://github.com/terminatorul/NvStrapsReBar/assets/2162540/53b7d155-e472-48e1-a544-b4c6583403e0)
> ![grafik](https://github.com/terminatorul/NvStrapsReBar/assets/2162540/f2bfcfa4-79cb-43e9-8dd4-ed32a8d34b3d)
> ![grafik](https://github.com/terminatorul/NvStrapsReBar/assets/2162540/0ca70949-450e-4ec7-89fc-785840c86ac1)
> ![grafik](https://github.com/terminatorul/NvStrapsReBar/assets/2162540/5629b82e-d765-485d-98ae-c733e3ab3687)

It is possible the NVIDIA App behaves like the NVIDIA Control Panel and always shows ReBAR disabled for Turing GPUs. Because ReBAR is officially supported only on Ampere (RTX 3000) and later GPUs.

@xCuri0 do you know why the firmware would hang on reboot ? 
Is there a restart notification I can hook, and then de-configure GPU ReBAR at that time ? Is there another script like the S3 Resume one ?

**Comment by terminatorul** [comment]
It is possible the NVIDIA App behaves like the NVIDIA Control Panel, and always shows ReBAR disabled for Turing GPUs. Because ReBAR is officially supported only on Ampere (RTX 3000) and later GPUs.

@xCuri0 do you know why the firmware would hang on reboot ? 
Is there a restart notification I can hook, and then de-configure GPU ReBAR at that time ? Is there another script like the S3 Resume one ?

**Comment by TheUntouchable** [comment]
Thanks for the quick answer :) 
Yeah I think the same, even the MSI Center is showing that rBAR ist active:
![grafik](https://github.com/terminatorul/NvStrapsReBar/assets/2162540/9ff5196c-dcdf-4856-be5d-bc6948d22d02)

Regarding that reboot thing, @vigilant-hylian had the same:

> I am having some issues however - my motherboard BIOS had two identical entries for the PCIE driver to add the .ffs file, and I added it to both. Now my PC will only start from cold, no reboots, no sleep resume, no hibernate resume. It'll not post, and the HDD light will remain on until i power off the PSU and restart. From full power down it works every time.
> 
> I have a feeling this might be a potential hurdle for some users as we iron out the kinks of patching compatible BIOSes - I'll do some testing at some point of adding the .ffs to one entry, then the other, and see if ReBAR continues to work and it resolves my issues. For now I can deal with cold boot :P

But in my case I had only one entry in the UEFI file.

**Comment by terminatorul** [comment]
@vigilant-hylian Did you try to add the NvStrapsReBar driver only once to the firmware image ?

Can you try the new version v0.3 (which fixes sleep) and say if you still have the resume from sleep issue ?

Do you have anything new about the reboot issue ? Are you stil dealing with it ?

**Comment by xCuri0** [comment]
you should use MMTool V5 on recent motherboards like this btw, using MMTOOL V4 might cause issues

**Comment by TheUntouchable** [comment]
I jus flashed the image I had created with the "MMTool Aptio 5.02.0024 Mod" and its exactly the same behaivour. Also rested all settings and tried it without any modification, same. Beforehand I deactivated BAR via "NvStrapsReBar.exe".
Will now try to flash the unmodified uefi, hopefully that will fix it again :D

**Comment by TheUntouchable** [comment]
Update: Even with the unmodified image from MSI directly I have now this behaivour :( Is there something I can reset with the "NvStrapsReBar.exe" in addition to disable BAR?
![grafik](https://github.com/terminatorul/NvStrapsReBar/assets/2162540/12e07027-3471-445d-978e-f283eae45e0b)

**Comment by TheUntouchable** [comment]
Update: I just realized that the CPU error LED on my motherboard is lit while my PC is freezing up

**Comment by terminatorul** [comment]
> Is there something I can reset with the "NvStrapsReBar.exe" in addition to disable BAR?

No, NvStrapsReBar does not modifiy the Setup variable.

Can you flash the original version of UEFI the board had pre-installed ? Or some older version of the firmware ?
Did you use MSI Live Update under Windows,  or M-FLASH from UEFI Setup ? Can you switch from one to the other ?
Can you boot from a Linux installation drive, to see if it does the same on shutdown ?

**Comment by TheUntouchable** [comment]
Okay, I just had a really hard time 😄.
I already tried to downgrade the UEFI as I had the feeling that the flashing with the same version over the modified version was just too fast...
But apparently I got a version which doesn't support my CPU yet and so I couldn't boot at all😄
Luckily I still had my old x2700 somewhere, but I have a fully water cooled cycle, so yeah....
I was able to update the uefi back to the latest version and now it seems I can reboot again with no problems, at least with the old CPU. Will have to replace the cpus again later..

**Comment by TheUntouchable** [comment]
Yes... I don't think it was my day... The PC is now running again, but the reboot problem is still there and the LED controller on my mainboard seems to be broken as well.

**Comment by terminatorul** [comment]
So can you still flash an older version (with support for your CPU), then use MSI Live Update (instead of M-FLASH) to flash the latest version again ?

**Comment by TheUntouchable** [comment]
I will try that out later :)

**Comment by TheUntouchable** [comment]
Update: Ok, I installed an earlier version via M-FLASH without any problems. Nevertheless, the LEDs on my graphics card are blinking like crazy, the LEDs on the mobo still don't light up and the PC still freezes after/while a reboot with red CPU Led on.

Unfortunately I can't flash the latest uefi version via Live Update as it has been marked as beta for 7 months :(

**Comment by terminatorul** [comment]
At any point did you try to clear CMOS and then manually load default / optimized values ?

**Comment by TheUntouchable** [comment]
My board has a dedicated button for that.. And I can't tell you how often I pressed it the last days :) Also I cleared the settings in the UEFI to default each time a did a flash.
